### PR TITLE
porter 1.1.0 (new formula)

### PR DESCRIPTION
**Pull request by SpiffyEight77**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/p/porter.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    cd "cmd/porter" do
+      ldflags = %W[
+        -s
+        -X get.porter.sh/porter/pkg.Version=#{version}
+        -X get.porter.sh/porter/pkg.Commit=#{git_head}
+      ]
+
+      system "go", "build", *std_go_args(ldflags:)
+      generate_completions_from_executable(bin/"porter", "completion")
+      system "#{bin}/porter", "mixin", "install", "exec", "--version", "v#{version}"
```
Comment (chenrui333) [inline Formula/p/porter.rb:23]:
```suggestion
      system bin/"porter", "mixin", "install", "exec", "--version", "v#{version}"
```

File: Formula/p/porter.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+  depends_on "go" => :build
+
+  def install
+    git_head = Utils.git_short_head
+    cd "cmd/porter" do
+      ldflags = %W[
+        -s
+        -X get.porter.sh/porter/pkg.Version=#{version}
+        -X get.porter.sh/porter/pkg.Commit=#{git_head}
```
Comment (chenrui333) [inline Formula/p/porter.rb:18]:
```suggestion
        -X get.porter.sh/porter/pkg.Commit=#{tap.user}
```

File: Formula/p/porter.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  url "https://github.com/getporter/porter.git",
+      tag:      "v1.0.17",
+      revision: "605e399737c5c89f0ce692c6aa3d9e6211250c95"
+  license "Apache-2.0"
+  head "https://github.com/getporter/porter.git", branch: "main"
+
+  depends_on "go" => :build
+
+  def install
+    git_head = Utils.git_short_head
```
Comment (chenrui333) [inline Formula/p/porter.rb:13]:
```suggestion
```

File: Formula/p/porter.rb [hunk 4]
```diff
@@ -0,0 +1,36 @@
+class Porter < Formula
+  desc "App artifacts, tools, configs, and logic packaged as distributable installer"
+  homepage "https://porter.sh"
+  url "https://github.com/getporter/porter.git",
+      tag:      "v1.0.17",
+      revision: "605e399737c5c89f0ce692c6aa3d9e6211250c95"
```
Comment (chenrui333) [inline Formula/p/porter.rb:6]: let's just use tarbal

File: Formula/p/porter.rb
Comment (chenrui333) [inline Formula/p/porter.rb:1]: we need a livecheck as well

image

Comment (SpiffyEight77) [reply Formula/p/porter.rb:1]: @chenrui333 

I've updated to the latest version, v1.1.0. 🙇🏻‍♂️

Could you review this PR again when you are available.

image

**Pull request closed without merging**